python/cuvs/CMakeLists.txt

```diff
@@ -70,8 +73,27 @@ if(NOT cuvs_FOUND)
   set(CUDA_STATIC_MATH_LIBRARIES ON)
   set(CUVS_USE_RAFT_STATIC ON)
 
+  if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL 12.0)
+    set(CUDA_STATIC_MATH_LIBRARIES OFF)
+  elseif(USE_CUDA_MATH_WHEELS)
+    message(FATAL_ERROR "Cannot use CUDA math wheels with CUDA < 12.0")
+  endif()
+
   add_subdirectory(../../cpp cuvs-cpp EXCLUDE_FROM_ALL)
 
+  if(NOT CUDA_STATIC_MATH_LIBRARIES AND USE_CUDA_MATH_WHEELS)
+    set(rpaths
+      "$ORIGIN/../nvidia/cublas/lib"
+      "$ORIGIN/../nvidia/cufft/lib"
+      "$ORIGIN/../nvidia/curand/lib"
+      "$ORIGIN/../nvidia/cusolver/lib"
+      "$ORIGIN/../nvidia/cusparse/lib"
+      "$ORIGIN/../nvidia/nvjitlink/lib"
+    )
```

Comment:
They are needed, because we have indirect dependencies on them. See https://github.com/rapidsai/cuml/pull/5966#discussion_r1714346964 - we tried taking them out and it caused a failure.

Reply:
Actually let me see if `cufft` is needed. If not, we may be able to take it out.

Reply:
cuML needs cufft here https://github.com/rapidsai/cuml/blob/branch-24.10/cpp/src/tsne/fft_tsne.cuh

Can we push this dependency down to cuML?

Reply:
@jameslamb is right that `cufft` was needlessly copied from the cuML PR. I've taken it out and we will soon see if it works without it.